pennylane/workflow/jacobian_products.py

```diff
@@ -678,7 +678,14 @@ def compute_vjp(self, tapes: Batch, dy: Tuple[Tuple[TensorLike]]) -> Tuple:
             logger.debug("compute_vjp called with (%s, %s)", tapes, dy)
         numpy_tapes = tuple(qml.transforms.convert_to_numpy_parameters(t) for t in tapes)
         dy = qml.math.unwrap(dy)
-        return self._device.compute_vjp(numpy_tapes, dy, self._execution_config)
+        vjps = self._device.compute_vjp(numpy_tapes, dy, self._execution_config)
+        res = []
+        for t, r in zip(tapes, vjps):
+            if len(t.trainable_params) == 1 and qml.math.shape(r) == ():
+                res.append((r,))
+            else:
+                res.append(r)
+        return res
```

Comment:
I'm not particularly fond of this as it's doing the job of the device. We've specified the return type specification quite clearly, and devices should obey that. I've added this for now for the sake of getting the tests to pass, but long term I think it's better to update the Lightning device(s) to fix the return type.

Reply:
Talked offline. We will leave this as is for now, update lightning to correctly handle vjp return types, and then remove this logic after that PR is merged.